src/Microsoft.ML.Core/CommandLine/CmdParser.cs

```diff
@@ -477,6 +477,11 @@ private static ArgumentInfo GetArgumentInfo(Type type, object defaults)
                         throw Contracts.Except("Duplicate default argument '{0}' vs '{1}'", def.LongName, field.Name);
 
                     string name = ArgCase(field.Name);
+
+                    string attrName = null;
+                    if (attr.Name != null && attr.Name.ToLowerInvariant() != name.ToLowerInvariant())
+                        attrName = attr.Name;
```

Comment:
My expectation is that the logic should be the same as for entry-points... if `Name` *is* set, then we would not use the field name from the user's perspective at all.